File: backend/src/Skidjakt.Core/Entities/Deal.cs
namespace Skidjakt.Core.Entities;

public class Deal
{
	public int Id { get; set; }
	public required string Agency { get; set; }
	public required string ExternalId { get; set; }
	public required string SourceUrl { get; set; }
	public required string Destination { get; set; }
	public required string Country { get; set; }
	public DateOnly? DepartureDate { get; set; }
	public DateOnly? ReturnDate { get; set; }
	public int? WeekNumber { get; set; }
	public int DurationNights { get; set; }
	public string? TransportType { get; set; }
	public string? DepartureCity { get; set; }
	public string? AccommodationName { get; set; }
	public string? AccommodationType { get; set; }
	public int? StarRating { get; set; }
	public required int PricePerPerson { get; set; }
	public int? OriginalPrice { get; set; }
	public int? DiscountAmount { get; set; }
	public bool IncludesFlight { get; set; }
	public bool IncludesLiftPass { get; set; }
	public bool IncludesMeals { get; set; }
	public bool IncludesTransfer { get; set; }
	public string? MealDescription { get; set; }
	public int? LiftPassDays { get; set; }
	public DateTime FirstSeen { get; set; }
	public DateTime LastSeen { get; set; }
	public bool IsActive { get; set; } = true;
}

File: frontend/src/types/deal.ts
export interface Deal {
  id: number;
  agency: string;
  externalId: string;
  sourceUrl: string;
  destination: string;
  country: string;
  departureDate: string | null;
  returnDate: string | null;
  weekNumber: number | null;
  durationNights: number;
  transportType: string | null;
  departureCity: string | null;
  accommodationName: string | null;
  accommodationType: string | null;
  starRating: number | null;
  pricePerPerson: number;
  originalPrice: number | null;
  discountAmount: number | null;
  includesFlight: boolean;
  includesLiftPass: boolean;
  includesMeals: boolean;
  includesTransfer: boolean;
  mealDescription: string | null;
  liftPassDays: number | null;
  firstSeen: string;
  lastSeen: string;
  isActive: boolean;
}

export interface DealQuery {
  search?: string;
  agencies?: string[];
  countries?: string[];
  destinations?: string[];
  minPrice?: number;
  maxPrice?: number;
  fromDate?: string;
  toDate?: string;
  transportTypes?: string[];
  sortBy?: string;
  page?: number;
  pageSize?: number;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface DealFilterOptions {
  agencies: string[];
  countries: string[];
  destinations: string[];
  transportTypes: string[];
  minPrice: number;
  maxPrice: number;
}

export interface DealStats {
  totalDeals: number;
  dealsPerAgency: Record<string, number>;
  averagePrice: number;
  lowestPrice: number;
  lastUpdated: string | null;
}
